service/handler_combinedKey.go

```diff
@@ -0,0 +1,103 @@
+package service
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/gorilla/mux"
+	"github.com/medibloc/panacea-oracle/crypto"
+	"github.com/medibloc/panacea-oracle/server/middleware"
+	log "github.com/sirupsen/logrus"
+)
+
+type Response struct {
+	EncryptedCombinedKey []byte `json:"encrypted-combined-key"`
+}
+
+func (svc *Service) GetCombinedKey(w http.ResponseWriter, r *http.Request) {
+	queryClient := svc.QueryClient()
+	oraclePrivKey := svc.OraclePrivKey()
+
+	dealIDStr := mux.Vars(r)["dealId"]
+	dealID, err := strconv.ParseUint(dealIDStr, 10, 64)
+	if err != nil {
+		log.Errorf("failed to parse deal ID: %s", err.Error())
+		http.Error(w, "failed to parse deal ID", http.StatusBadRequest)
+		return
+	}
+
+	dataHashStr := mux.Vars(r)["dataHash"]
+	dataHash, err := hex.DecodeString(dataHashStr)
+	if err != nil {
+		log.Errorf("failed to decode dataHash: %s", err.Error())
+		http.Error(w, "failed to decode dataHash", http.StatusBadRequest)
+		return
+	}
+	var dataHash32 [sha256.Size]byte
+	copy(dataHash32[:], dataHash)
```

Comment:
I copied and pasted the `getComnbinedKey` function from PR #20. In order to use it as an input parameter at this time, I converted it to [32] bytes, but now it seems that it would be better to unify it as [] byte type. I fixed it in [108af49](https://github.com/medibloc/panacea-oracle/pull/22/commits/108af4927c7714fb5570772be66060ac2ca676cb).